Scripts/build/Jenkins/Jenkinsfile

```diff
@@ -344,65 +340,85 @@ def PreBuildCommonSteps(Map pipelineConfig, String repositoryName, String projec
                 script: 'python/get_python.bat'
         }
 
-        if(env.CLEAN_OUTPUT_DIRECTORY?.toBoolean() || env.CLEAN_ASSETS?.toBoolean()) {
-            def command = "${pipelineConfig.BUILD_ENTRY_POINT} --platform ${platform} --type clean"
-            if (env.IS_UNIX) {
-                sh label: "Running ${platform} clean",
-                   script: "${pipelineConfig.PYTHON_DIR}/python.sh -u ${command}"
-            } else {
-                bat label: "Running ${platform} clean",
-                    script: "${pipelineConfig.PYTHON_DIR}/python.cmd -u ${command}".replace('/','\\')
-            }
-        }
+        // Always run the clean step, the scripts detect what variables were set, but it also cleans if
+        // the NODE_LABEL has changed
+        def command = "${pipelineConfig.PYTHON_DIR}/python"
+        if(env.IS_UNIX) command += '.sh'
+        else command += '.cmd'
+        command += " -u ${pipelineConfig.BUILD_ENTRY_POINT} --platform ${platform} --type clean"
+        palSh(command, "Running ${platform} clean")
     }
 }
 
-def Build(Map options, String platform, String type, String workspace) {
+def Build(Map pipelineConfig, String platform, String type, String workspace) {
     // If EXECUTE_FROM_PROJECT is defined, we execute the script from the project instead of from the engine
     // In both cases, the scripts are in the engine, is just what the current dir is and how we get to the scripts
     def currentDir = "${workspace}/${ENGINE_REPOSITORY_NAME}"
     def pathToEngine = ""
-    def scriptExt = ""
-    if (env.IS_UNIX) {
-        scriptExt = ".sh"
-    }
-    if (env.EXECUTE_FROM_PROJECT?.toBoolean()) {
-        currentDir = "${workspace}/${PROJECT_REPOSITORY_NAME}"
-        pathToEngine = "../${ENGINE_REPOSITORY_NAME}/"
-    }
-    else {
-        dir("${currentDir}") {
-            palSh("scripts/o3de${scriptExt} register --project-path ${workspace}/${PROJECT_REPOSITORY_NAME}", "Registering project ${PROJECT_REPOSITORY_NAME}") // o3de.sh will work under Windows in a Cygwin environment
+
+    timeout(time: env.TIMEOUT, unit: 'MINUTES', activity: true) {
+        def command = "${pipelineConfig.PYTHON_DIR}/python"
+        def ext = ''
+        if(env.IS_UNIX) {
+            command += '.sh'
+            ext = '.sh'
+        }
+        else command += '.cmd'
+
```

Comment:
It might be safer to set 'ext' here as well for windows, in case there was a command with .exe, .bat. and .cmd. (The priority of extensions when running the command is based on the 'PATHEXT' environment variable

Reply:
In this case, its only to run `o3de.bat` for project registration, but we can set that specifically.